pkg/guacrest/openapi.yaml

```diff
@@ -77,41 +79,74 @@ paths:
 
 
 components:
+  parameters:
+    PaginationSpec:
+      name: PaginationSpec 
+      in: query
+      description: >
+        The pagination configuration for the query.
+          * 'PageSize' specifies the number of results returned
+          * 'Cursor' is returned by previous calls and specifies what page to return
+      required: false
+      schema:
+        type: object
+        properties:
+          PageSize: 
+            type: integer 
+          Cursor: 
+            type: string
   schemas:
     Purl:
       type: string
+    PaginationInfo:
+      type: object
+      required:
+        - NextCursor
```

Comment:
Should `NextCursor` be required?

I am asking because `NextCursor` is being set to an empty string on the last page of the REST output which clients would have to interpret correctly.

If `NextCursor` wasn’t required and could be nullable, then we could set `NextCursor` to nil. 

i.e.

``` go
end := start + uint64(pagesize) // end is exclusive
var nextCursor *string // Use a pointer to string to allow nil value
if end < inputLength {
    cursorStr := cursorFromIndex(end)
    nextCursor = &cursorFromIndex(end) // Only set cursorFromIndex if there are more pages
}

info := models.PaginationInfo{
    NextCursor: nextCursor, // This can now be nil if there are no more items
    TotalCount: PointerOf(len(lst)),
}
```

And, for the client side handling
* When `NextCursor` isn’t provided in the response or is `nil`, then they know that there are no more pages to fetch.
* If `NetCursor` isn’t provided from the used, we could interpret this as a request for the first page of data.

If we did this we would also have to provide comments as to the `NextCursor` being `nil`.

Reply:
@mdeicas just need to address this comment and PR is ready to merge.